src/openrct2/network/Socket.cpp

```diff
@@ -77,7 +77,123 @@ class SocketException : public std::runtime_error
     }
 };
 
-class TcpSocket final : public ITcpSocket
+class NetworkEndpoint final : public INetworkEndpoint
+{
+private:
+    sockaddr _address{};
+    socklen_t _addressLen{};
+
+public:
+    NetworkEndpoint()
+    {
+    }
+
+    NetworkEndpoint(const sockaddr* address, socklen_t addressLen)
+    {
+        std::memcpy(&_address, address, addressLen);
+        _addressLen = addressLen;
+    }
+
+    const sockaddr& GetAddress() const
+    {
+        return _address;
+    }
+
+    socklen_t GetAddressLen() const
+    {
+        return _addressLen;
+    }
+
+    int32_t GetPort() const
+    {
+        if (_address.sa_family == AF_INET)
+        {
+            return ((sockaddr_in*)&_address)->sin_port;
+        }
+        else
+        {
+            return ((sockaddr_in6*)&_address)->sin6_port;
+        }
+    }
+
+    std::string GetHostname() const override
+    {
+        char hostname[256];
```

Comment:
```suggestion
        char hostname[256]{};
```
Not sure if thats even possible but better better safe than sorry.